### DISCUSSION TITLE
Absolutely awesome. How does this work?

### DISCUSSION THREAD

**Discussion by Neeqstock** [Coyote-A/ultimate-upscale-for-automatic1111#41] [Q&A]
I effectively got better results from SD Upscale, and definitely getting less artifacts. Also, I absolutely like having more control on final resolution, seam correction method, etc.
How does this technically work? In how it's different from SD Upscale? Is it somewhat more aware of the contents of the whole image? I noticed it requires less tiles, so i guess the algorithm should be more "content aware", and at the same time faster.
And lastly, how can you tell me the difference between types "Linear, Chess, None" in the upscsaler settings?

I'm usually working with 768x768 tiles, 1536x1536 final image (2x upscale), 0.3-0.4 denoising strength, DPM++ SDE (sometimes Karras), 20-25 steps, ERSGAN_4x and I get very good results.

Thanks for your work!

**Comment by Coyote-A** [comment]
It work in 3 steps:
1) upscale image with selected upscaler
2) redraw image with many inpaints. it use tile size to crop area for inpaint. padding is used to see content of neighboring tiles. mask blur is used to blur tile mask. it allows to smooth slightly seam.
3) seam fix passes. run extra inpaints on seams with gradient masks. it fix artifacts on seams and makes the mosaic of tiles less visible.

> And lastly, how can you tell me the difference between types "Linear, Chess, None" in the upscsaler settings?

linear - all tiles processed one by one. column by column, row by row
chess - all tiles processd in chess order. decrease chance of getting artifacts on seams.
none - redraw is disabled. use it when run generation without seam fix, see visible overlays or artifacts on seams and want to run just seam pass. put upscaled image as a source before
